recursion.rb

```diff
@@ -1,17 +1,82 @@
+# Lisa Rolczynski
 
+# RC: n! = n * (n-1)!
+# BC: 1! = 1
+# BC: 0! = 1
 def fact(n)
+  if n == 1 || n == 0
+    return 1
+  end
+
+  return n * fact(n-1)
 end
 
+# BC: fib(1) = 1
+# BC: fib(2) = 1
 def fib(n)
+  if n == 1 || n == 2
+    return 1
+  end
+
+  fib(n-1) + fib(n-2)
 end
 
+# palindromes
+# BC: if there's 1 or 0 letters left, return true
 def pal(s)
+  s_array = s.chars
+  start_index = 0
+  end_index = s_array.length - 1
+
+  pal_helper(s_array, start_index, end_index)
 end
 
+def pal_helper(s_array, start_index, end_index)
```

Comment:
This works, but it's unnecessary to pass the start and end value and/or to use a helper method -- though it works! Check out @cotarg 's solution which is pretty clean and easily readable.